### PR TITLE
Infinite scroll of posts

### LINKED ISSUE #5: Feat: Create an infinite scroll of posts
This feature shows the implementation of an infinite scrolling of posts generated from the Placeholder Posts API.

### PR DESCRIPTION
**Pull request by motuncoded**
In this change, I created a Posts.tsx file that can allow infinite scroll of posts with loading when navigating down the posts.

closes #5

### CODE REVIEW

File: pages/Posts.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  id: number;
+  title: string;
+  body: string;
+  userId: number;
+};
+
+type Error = {
+  message: string;
+};
+type PageNumber = number;
```
Comment (Asylcreek) [inline pages/Posts.tsx]: You did not need to create a type alias for this.

Typing line 26 as `number` would have been enough

File: pages/Posts.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+
+export default function Posts() {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+  const [page, setPage] = useState<PageNumber>(1);
+  const [hasMore, setHasMore] = useState(true);
```
Comment (Asylcreek) [inline pages/Posts.tsx]: I don't think `true` should be the default

File: pages/Posts.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      setLoading(true);
+      const response = await fetch(`${POSTS_URL}${page}`);
+      if (!response.ok) {
+        throw new Error("Failed to fetch posts");
+      }
+      const posts = await response.json();
+      setPosts((prev)=> [...prev, ...posts]);
+      if (posts.length < 10) {
+        setHasMore(false);
+      }
```
Comment (Asylcreek) [inline pages/Posts.tsx]: Flip this condition around

File: pages/Posts.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    }, 1000);
+  }, [page]);
+
+
+
+
+
+  const handleScroll = () => {
+    if (window.innerHeight + document.documentElement.scrollTop + 1 >=
+      document.documentElement.scrollHeight
```
Comment (Asylcreek) [inline pages/Posts.tsx]: Shouldn't these be `document.body` instead?

File: pages/Posts.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    setTimeout(() => {
+      fetchPosts();
+    }, 1000);
+  }, [page]);
+
+
+
+
+
+  const handleScroll = () => {
```
Comment (Asylcreek) [inline pages/Posts.tsx]: Move this function inside the useEffect below

File: pages/Posts.tsx [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    ) {
+      if (hasMore) {
+        setPage((prev) => prev + 1);
+      }
+    }
+  };
+  useEffect(() => {
+    window.addEventListener("scroll", handleScroll);
+    return () => window.removeEventListener("scroll", handleScroll);
+  });
```
Comment (Asylcreek) [inline pages/Posts.tsx]:
```suggestion
  }, []);
```

**Review by Asylcreek — CHANGES_REQUESTED** [review]

**Review by Asylcreek — DISMISSED** [review]

File: pages/Posts.tsx
Comment (Asylcreek) [inline pages/Posts.tsx:1]: Why did you create this in a separate file?

Do you realize that `/` and `/posts` would practically be showing the same things?

Comment (motuncoded) [reply pages/Posts.tsx:1]: ok

Comment (motuncoded) [reply pages/Posts.tsx:1]: I noticed that the posts rendered twice according to id and I have been trying to fix it and it seems unattainable, can you please recommend something
copt

Comment (Asylcreek) [reply pages/Posts.tsx:1]: What exactly do you want to happen?

Which route do you want your users to access posts from?

**Review by Asylcreek — APPROVED** [review]

**Pull request merged**